packages/ember-intl/addon/services/intl.ts

```diff
@@ -0,0 +1,397 @@
+import EventEmitter from 'eventemitter3';
+
+import type {
+  FormatDateParameters,
+  FormatListParameters,
+  FormatMessageParameters,
+  FormatNumberParameters,
+  FormatRelativeParameters,
+  FormatTimeParameters,
+  IntlShape,
+  OnErrorFn,
+} from '../-private/formatjs/index';
+import {
+  createIntl,
+  createIntlCache,
+  formatDate,
+  formatList,
+  formatMessage,
+  formatNumber,
+  formatRelative,
+  formatTime,
+} from '../-private/formatjs/index';
+import { escapeFormatMessageOptions } from '../-private/utils/escape-format-message-options';
+import { getDOM } from '../-private/utils/get-dom';
+import {
+  convertToArray,
+  convertToString,
+  hasLocaleChanged,
+} from '../-private/utils/locale';
+import { flattenKeys, type Translations } from '../-private/utils/translations';
+import translations from '../translations';
+
+type OnFormatjsError = (error: Parameters<OnErrorFn>[0]) => void;
+
+export default class IntlService extends Service {
+  @tracked private _intls: Record<string, IntlShape> = {};
+  @tracked private _locale?: string[];
+
+  private _cache = createIntlCache();
+  private _eventEmitter = new EventEmitter();
+  private _formats?: Record<string, unknown>;
+  private _timer?: EmberRunTimer;
+
+  get locales(): string[] {
+    return Object.keys(this._intls);
+  }
+
+  get primaryLocale(): string | undefined {
+    if (!this._locale) {
+      return;
+    }
+
+    return this._locale[0];
+  }
+
+  constructor() {
+    // eslint-disable-next-line prefer-rest-params
+    super(...arguments);
+
+    if (!this._formats) {
+      // @ts-expect-error: Property 'resolveRegistration' does not exist on type 'Owner'
+      this._formats = getOwner(this).resolveRegistration('formats:main') ?? {};
+    }
+
+    // Hydrate
+    translations.forEach(([locale, translations]: [string, Translations]) => {
+      this.addTranslations(locale, translations);
+    });
+  }
+
+  willDestroy() {
+    super.willDestroy();
+
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore: Argument of type 'Timer | undefined' is not assignable to parameter of type 'EmberRunTimer | undefined'
+    // eslint-disable-next-line ember/no-runloop
+    cancel(this._timer);
+  }
+
+  addTranslations(locale: string, translations: Translations) {
+    const messages = flattenKeys(translations);
+
+    this.updateIntl(locale, messages);
+  }
+
+  exists(key: string, locale?: string | string[]): boolean {
+    const candidates = locale ? convertToArray(locale) : this._locale!;
+
+    return candidates.some((candidate) => {
+      const messages = this.getIntl(candidate)?.messages;
+
+      if (!messages) {
+        return false;
+      }
+
+      return key in messages;
+    });
+  }
+
+  formatDate(
+    value: FormatDateParameters[0] | undefined | null,
+    options?: FormatDateParameters[1] & {
+      locale?: string;
+    },
+  ): string {
+    if (value === undefined || value === null) {
+      return '';
+    }
```

Comment:
**breaking**

For consistency, all `format*()` methods will return an empty string when `value` is `undefined` or `null`.

(`@formatjs/intl` may or may not accept these values.)